vortex-array/src/array/chunked/canonical.rs

```diff
@@ -47,9 +49,14 @@ pub(crate) fn try_canonicalize_chunks(
         // Extension arrays wrap an internal storage array, which can hold a ChunkedArray until
         // it is safe to unpack them.
         DType::Extension(ext_dtype, _) => {
+            let storage_chunks: Vec<Array> = chunks
+                .iter()
+                .map(|chunk| ExtensionArray::try_from(chunk).unwrap().storage())
+                .collect();
+            let storage_dtype = storage_chunks.first().unwrap().dtype().clone();
             let ext_array = ExtensionArray::new(
                 ext_dtype.clone(),
-                ChunkedArray::try_new(chunks, dtype.clone())?.into_array(),
+                ChunkedArray::try_new(storage_chunks, storage_dtype)?.into_array(),
```

Comment:
hmm, maybe this is enough. Not sure though since it behaves differently than any other canonical type